tools/db_stress.cc

```diff
@@ -1756,7 +1767,54 @@ class StressTest {
 
       MaybeClearOneColumnFamily(thread);
 
-#ifndef ROCKSDB_LITE  // Lite does not support GetColumnFamilyMetaData
+#ifndef ROCKSDB_LITE
+      if (FLAGS_checkpoint_one_in > 0 &&
+          thread->rand.Uniform(FLAGS_checkpoint_one_in) == 0) {
+        std::string checkpoint_dir =
+            FLAGS_db + "/.checkpoint" + ToString(thread->tid);
+        Checkpoint* checkpoint;
+        Status s = Checkpoint::Create(db_, &checkpoint);
+        if (s.ok()) {
+          s = checkpoint->CreateCheckpoint(checkpoint_dir);
+        }
+        std::vector<std::string> files;
+        if (s.ok()) {
+          s = FLAGS_env->GetChildren(checkpoint_dir, &files);
+        }
+        size_t file_idx = 0;
+        while (s.ok() && file_idx < files.size()) {
```

Comment:
Must we delete all files before deleting the dir?

Reply:
sadly yeah, our `DeleteDir` uses `rmdir(3)` which requires the directory to be empty :/.

Reply:
btw I am thinking about following your pattern and making `TestBackup`, `TestCheckpoint`, etc., for satisfying all the "*_one_in" flags.

Reply:
Are `TestBackup` and `TestCheckpoint` expected to be overridden in subclasses? Do you expect subclasses to override them?

Reply:
Actually no they most likely won't be overridden, so I'm curious if you think that is still the right approach, or have any other ideas.